### PR TITLE
Handle address/EC2 instance exhaustion, break instance delete into batches

### PR DESCRIPTION
**Pull request by bretttegartms**
Autoscaler fixes.

### CODE REVIEW

File: lib/python/treadmill_aws/hostmanager.py [hunk 1]
```diff
@@ -5,6 +5,8 @@
 import time
 import yaml
 
+from botocore.exceptions import ClientError
```
Comment (andreikeis) [inline lib/python/treadmill_aws/hostmanager.py]: we do not import classes, - from botocore import exceptions as botoexc something like that.

File: lib/python/treadmill_aws/hostmanager.py [hunk 2]
```diff
@@ -134,7 +152,10 @@ def delete_hosts(ec2_conn, ipa_client, hostnames):
             _LOGGER.debug('Host not found: %s', hostname)
 
     _LOGGER.debug('Delete instances: %r', hostnames)
-    ec2client.delete_instances(ec2_conn, hostnames=hostnames)
+    ec2_chunks = len(hostnames) // 50 + 1
+    for i in range(ec2_chunks):
+        batch = hostnames[i * 50:(i + 1) * 50]
```
Comment (andreikeis) [inline lib/python/treadmill_aws/hostmanager.py]: magic number? Make it _DELETE_BATCH_SIZE and put comment why it is needed.

Comment (bretttegartms) [reply lib/python/treadmill_aws/hostmanager.py]: Fixed in subsequent PR.

File: lib/python/treadmill_aws/hostmanager.py (hunk 1 above)
Comment (bretttegartms) [reply lib/python/treadmill_aws/hostmanager.py]: Fixed in subsequent PR.

**Pull request merged**